src/TrieSet.mo

```diff
@@ -70,4 +71,19 @@ public type Set<T> = Trie.Trie<T,()>;
     s3
   };
 
+  //// Construct a set from an array.
+  public func fromArray<T>(arr: [T], elemHash: T -> Hash, eq: (T, T) -> Bool): Set<T> {
+    let assocList = List.zipWith<T, (), (Trie.Key<T>, ())>(
+      List.fromArray(arr),
+      List.tabulate<()>(arr.size(), func(_: Nat): (){}),
+      func(t: T, _: ()): (Trie.Key<T>, ()) { ({key=t; hash=elemHash(t)}, ()) }
+    );
+    Trie.fromList<T, ()>(null, assocList, 0)
+  };
+
+  //// Returns the set as an array.
+  public func toArray<T>(s: Set<T>): [T] {
+    Trie.toArray<T, (), T>(s, func (t: T, _: ()): T { t })
```

Comment:
```suggestion
    Trie.toArray(s, func (t: T, _: ()): T { t })
```

Motoko actually has better inference for these situations now. You can either annotate the lambda or `Trie.toArray` here but don't need both anymore.